### ISSUE TITLE
Create and update a transaction

### ISSUE THREAD

**Issue by aksoftware98** [aksoftware98/ak-expenses-tracker#4]
The user must be able to insert a new transaction or update an already existing one.

The transaction must be associated with a **wallet** and it could be either a **Income** or **Outcome**, it also has a list of tags that simplifies the process of searching, grouping, and listing the transactions. 

` Transaction can have a file or list of files (documents) associated with it. `

## Tasks to be done
 - [ ] Implement the insert or update functionality in the repository
 - [ ] Create the required DTOs
 - [ ] Implment the logic to upload a file to Azure Blob storage
 - [ ] Create an Azure Function to upload and validate an uploaded file using Azure Cognitve Services.
 - [ ] Create an Azure Function to insert or update a transaction.

**Comment by Gareth064** [comment]
A concept that I am yet see any other app, of this type, try to implement is "transfers" between accounts/wallets.
It is a hard one to do (which is why they never really do it)

But why I personally want it is because I am always moving money between **MY** bank accounts, and all my apps report that movement of money as Income or Expenses. So when I run a report for a month it thinks my Income is a lot higher than it really was. Lets say for example I have 1 income from my job of £3000 and it goes into one account. Then I move £1000 to another account. These other apps record that £1000 going into my other account as income, so now the report states £4000 income for that month.

One of the reasons it's hard to do is because there is normally nothing on a transaction of the receiving account/wallet to allow you automatically reconcile it with the same transaction of the outgoing account/wallet. So if you want this to look correct on reports, you normally have to do some sort of manual reconciliation.

But maybe with the use of Azure Cognitive Services, it could at least try to detect these transfers and possibly at least give you suggestions that maybe something is a transfer from one of your other accounts/wallets.

**Comment by aksoftware98** [comment]
@Gareth064 This is also a very great feature, I will implement that for sure because I also do the exact same thing. 

One thing regarding the applications to detect your accounts among the banks and wallets it's nearly impossible, because overall the banks globally despite all the technology you see on top, but basically, they are built on a broken technical infrastructure. 
And until now you cannot find a bank with a great, clear API, and even if there is one, it's only one bank and not all of them, and until the moment too many things happens manullay behind the scenes and this is why when I make a transfer from Europe to somewhere else it takes 5 days.
So basically what is going to happen in our app is that, after you make an actual transfer from your account and got the money in the other one, you can open the app and click the transfer feature from one wallet to the other and it will automatically creates an income transaction in the second bank and an outcome one in the first one. 

I think this will help to fix the issue.
